### PR TITLE
HPCC-15499 Redis Plugin: add redis command INCRBY

### PR DESCRIPTION
**Pull request by jamienoss**
Signed-off-by: James Noss james.noss@lexisnexis.com

### CODE REVIEW

**Comment by jamienoss** [comment]
@ghalliday please review.
@richardkchapman Here is the new rebased PR, can you amend the PR link in the Jira please.

**Comment by richardkchapman** [comment]
For future reference, there's generally no need (and it is preferable not to) open a new PR when rebasing. Rebase the original branch then force-push.

File: plugins/redis/redis.cpp [hunk 1]
```diff
@@ -167,7 +174,10 @@ protected :
     void assertConnection(const char * _msg);
     void assertConnectionWithCmdMsg(const char * cmd, const char * key = NULL);
     void fail(const char * cmd, const char * errmsg, const char * key = NULL);
-    void * redisCommand(redisContext * context, const char * format, ...);
+    void * redisCommand(const char * format, ...);
+    void strto(const char * str, double & ret, const char * key);
```
Comment (ghalliday) [inline plugins/redis/redis.cpp]: style: Generally convention is to either have output parameters first in the list or last in the list.  (I prefer them 1st, google style is for them to be last.)

File: plugins/redis/redis.cpp [hunk 2]
```diff
@@ -515,38 +525,68 @@ void Connection::assertConnection(const char * _msg)
...
+
+    appendExpire(cmd, expire);
+
+    OwnedReply reply = Reply::createReply(redisCommand(cmd.str(), key, strlen(key), value));
+    assertOnErrorWithCmdMsg(reply->query(), "SET", key);
+}
+void Connection::setReal(ICodeContext * ctx, const char * key, double value, unsigned expire)
+{
+    /*MORE: I really think that this should be 15 and that we are printing 1 to many (i.e. 16) significant digits for results.
+     * This may be related to https://track.hpccsystems.com/browse/HPCC-12177 "Instability in floating point calculations for statistical functions".
```
Comment (ghalliday) [inline plugins/redis/redis.cpp]: I don't think this comment is relevant. It isn't related to instability in numeric calculations. And that issue is also not related to storing results as text.

File: plugins/redis/redis.cpp [hunk 3]
```diff
@@ -591,10 +646,39 @@ template<class type> void SyncRGet(ICodeContext * ctx, const char * options, con
     Owned<Connection> master = Connection::createConnection(ctx, cachedConnection, options, database, password, _timeout, cacheConnections);
     master->get(ctx, key, returnSize, returnValue);
 }
+void Connection::strto(const char * str, double & ret, const char * key)
```
Comment (ghalliday) [inline plugins/redis/redis.cpp]: hyper-picky:  strto is a strange name since it begs the question to what.  fromstr would read much more naturally.

**Comment by ghalliday** [comment]
@jamienoss I would merge as is, but I think that comment needs changing/deleting first.

**Comment by jamienoss** [comment]
@ghalliday changes made, give the all ok and I can then quickly squash the last commit.

**Comment by jamienoss** [comment]
@ghalliday please merge, thanks.
